mp/src/game/server/env_player_surface_trigger.cpp

```diff
@@ -89,6 +93,11 @@ void CEnvPlayerSurfaceTrigger::PlayerSurfaceChanged( CBasePlayer *pPlayer, char
 	if ( m_bDisabled )
 		return;
 
+    if (pPlayer->m_chPreviousTextureType == gameMaterial)
+        return;
+
+    pPlayer->m_chPreviousTextureType = gameMaterial;
```

Comment:
Why not just set it?

Reply:
Turns out `m_chPreviousTextureType` doesn't serve any other purpose so I'll just remove it entirely.